### PR TITLE
Update: refactor reader/discover/helper

### PR DESCRIPTION
**Pull request by jhnstn**
I added some tests for the helper and did some refactoring as part of #7057

The behavior of the helper functions is the same aside from a few tweaks to the return types of a couple of the helpers.  Some of the predicate functions where returning `true`, `false` or `undefined`. In these cases I eliminated the `undefined` return types. There was also a _getter_ helper that returned `false`. I changed that to return `undefined` for the `false` case.

I also updated files where the `import` syntax was being used to import the helper.

**Testing** 
    
    npm run test-client client/reader/discover/

cc @blowery 

Test live: https://calypso.live/?branch=update/reader/refactor-discover-helper

### CODE REVIEW

File: client/reader/list-stream/missing.jsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import React from 'react';
  * Internal dependencies
  */
 import EmptyContent from 'components/empty-content';
-import discoverHelper from 'reader/discover/helper';
+import { isEnabled as discoverIsEnabled } from 'reader/discover/helper';
```
Comment (jhnstn) [inline client/reader/list-stream/missing.jsx]: I might make a second pass and rename this as 'isDiscoverEnabled` in the helper to clean up import syntax

**Comment by bluefuton** [comment]
Looks good to me @jhnstn.

I'll +1 your suggestion to rename `isEnabled` to `isDiscoverEnabled` in the helper.

**Comment by jhnstn** [comment]
@bluefuton I just updated the `isEnabled` bit.
Can I get one last review before merging?

**Comment by bluefuton** [comment]
Looks good to 🚢!

Thanks for adding tests :)

**Pull request merged**